test/Unosquare.Swan.Test/ObjectComparerTest.cs

```diff
@@ -11,76 +9,170 @@ namespace Unosquare.Swan.Test
-
-            Assert.IsTrue(ObjectComparer.AreStructsEqual(left, right));
+            [Test]
+            public void CompareEqualsObjects_ReturnsTrue()
+            {
+                var left = DateBasicJson.GetDateDefault();
+                var right = DateBasicJson.GetDateDefault();
+
+                Assert.IsTrue(ObjectComparer.AreObjectsEqual(left, right));
+            }
+
+            [Test]
+            public void CompareDifferentObjects_ReturnsFalse()
+            {
+                var left = BasicJson.GetDefault();
+                var right = new BasicJson();
+
+                Assert.IsFalse(ObjectComparer.AreObjectsEqual(left, right));
+            }
+
+            [Test]
+            public void WithDifferentObjects_ReturnsFalse()
+            {
+                List<object> leftListObject = new List<object>
+                {
+                    "ArCiGo",
+                    "ElCiGo",
+                    true
+                };
+
+                List<object> rightListObject = new List<object>
+                {
+                    "Néstor",
+                    "Florencia",
+                    1,
+                    false
+                };
+
+                Assert.IsFalse(ObjectComparer.AreObjectsEqual(leftListObject, rightListObject));
+            }
         }
-
-        [Test]
-        public void CompareDifferentStructsTest()
+        
+        [TestFixture]
+        public class AreStructsEqual : ObjectComparerTest
         {
-            var left = new SampleStruct() { Name = "PEPE", Value = 1 };
-            var right = new SampleStruct() { Name = "PEPE", Value = 2  };
-
-            Assert.IsFalse(ObjectComparer.AreStructsEqual(left, right));
+            [Test]
+            public void CompareEqualStructs_ReturnsTrue()
+            {
+                var leftStruct = new SampleStruct();
+                var rightStruct = new SampleStruct();
+
+                Assert.IsTrue(ObjectComparer.AreStructsEqual(leftStruct, rightStruct));
+            }
+
+            [Test]
+            public void CompareDifferentStructs_ReturnsFalse()
+            {
+                var leftStruct = new SampleStruct()
+                {
+                    Name = "ArCiGo",
+                    Value = 1
+                };
+
+                var rightStruct = new SampleStruct()
+                {
+                    Name = "Kadosh",
+                    Value = 2
+                };
+
+                Assert.IsFalse(ObjectComparer.AreStructsEqual(leftStruct, rightStruct));
+            }
         }
 
-        [Test]
-        public void CompareEqualsArrayTest()
+        [TestFixture]
+        public class AreArraysEqual : ObjectComparerTest
         {
-            var first = new[] { 1, 2, 3 };
-            var second = new[] { 1, 2, 3 };
-
-            Assert.IsTrue(ObjectComparer.AreEnumsEqual(first, second));
+            [Test]
+            public void CompareEqualArrays_ReturnsTrue()
```

Comment:
Maybe even remove the word "Compare" from the method name..so it would be in format "State_ExpectedResult"